src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -222,13 +226,9 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
   }
 
   private fun loadCommands() {
-    tryRestartingAgentIfNotRunning(project)
-    getInitializedServer(project).thenAccept { server ->
-      if (server == null) {
-        setRecipesPanelError()
-      }
+    CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
```

Comment:
I guess we can simplify the code here even more. Let's stick to the rule that it is the caller's responsibility to use a proper thread, hence:
1. remove `CodyAgentService.applyAgentOnBackgroundThread(project) { ... }` wrapping here
2. remove `ApplicationManager.getApplication().executeOnPooledThread { ... }` wrapping above (line 225, in `refreshRecipes`)
3. revamp `loadCommands` so it takes `agent` as a param
3. wrap `loadCommands(agent)` into `CodyAgentService.applyAgentOnBackgroundThread(project) { ... }` in line 225, `refeshRecipes`)

Reply:
1. I agree that we can simplify this code a bit
2. I do not agree with `Let's stick to the rule that it is the caller's responsibility to use a proper thread` as a general rule, it can lead to too many errors/omissions. That is, unless we explicitly wrap methods bory in something like CompletableFuture, but with a distinct subtypes for EDT and Background (which I think could work nicely but it's a bit additional work and I'm not sure if it's currently worth it).
3. I'm happy to discuss this on the standup or a dedicated call.

Reply:
ok, feel free to do it your way